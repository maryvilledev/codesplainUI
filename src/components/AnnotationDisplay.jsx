import React, { PropTypes } from 'react';
import MarkdownRenderer from 'react-markdown-renderer';
import RaisedButton from 'material-ui/RaisedButton';

import markdownRendererOptions from '../util/markdown-renderer-options';

const AnnotationDisplay = (props) => {
  const {
    annotation,
    closeAnnotation,
    editAnnotation,
  } = props;

  return (
    <div>
      <MarkdownRenderer
        markdown={annotation}
        options={markdownRendererOptions}
      />
      <RaisedButton
        label="Close"
        secondary
        onTouchTap={closeAnnotation}
      />
      <RaisedButton
        label="Edit"
        primary
        onTouchTap={editAnnotation}
      />
    </div>
  );
};

AnnotationDisplay.propTypes = {
  annotation: PropTypes.string.isRequired,
  closeAnnotation: PropTypes.func.isRequired,
  editAnnotation: PropTypes.func.isRequired,
};

export default AnnotationDisplay;
