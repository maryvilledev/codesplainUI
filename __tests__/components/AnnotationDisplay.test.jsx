import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

import AnnotationDisplay from '../../src/components/AnnotationDisplay';

const defaultProps = {
  annotation: '',
  closeAnnotation: jest.fn(),
  editAnnotation: jest.fn(),
  getNextAnnotation: jest.fn(),
  getPreviousAnnotation: jest.fn(),
  hasNextAnnotation: false,
  hasPrevAnnotation: false,
};

describe('<AnnotationDisplay />', () => {
  const muiTheme = getMuiTheme();
  const shallowWithContext = node => shallow(node, { context: { muiTheme } });

  describe('snapshots', () => {
    describe('when no annotations before/after current one exist', () => {
      it('matches snapshot', () => {
        const wrapper = shallowWithContext(<AnnotationDisplay {...defaultProps} />);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
      });
    });
    describe('when an annotation after current one exists', () => {
      it('matches snapshot', () => {
        const wrapper = shallowWithContext(
          <AnnotationDisplay
            {...defaultProps}
            hasNextAnnotation
          />,
        );
        expect(shallowToJson(wrapper)).toMatchSnapshot();
      });
    });
    describe('when an annotation before current one exists', () => {
      it('matches snapshot', () => {
        const wrapper = shallowWithContext(
          <AnnotationDisplay
            {...defaultProps}
            hasPrevAnnotation
          />,
         );
        expect(shallowToJson(wrapper)).toMatchSnapshot();
      });
    });
    describe('when annotation exists before/after current one', () => {
      it('matches snapshot', () => {
        const wrapper = shallowWithContext(
          <AnnotationDisplay
            {...defaultProps}
            hasNextAnnotation
            hasPrevAnnotation
          />,
         );
        expect(shallowToJson(wrapper)).toMatchSnapshot();
      });
    });
  });

  describe('prop: annotation', () => {
    it('is rendered into a MarkdownRenderer', () => {
      const annotation = 'Wubba lubba dub dub!';
      const wrapper = shallowWithContext(
        <AnnotationDisplay
          {...defaultProps}
          annotation={annotation}
        />,
      );
      expect(wrapper.find('MarkdownRenderer').prop('markdown')).toEqual(annotation);
    });
  });
});
