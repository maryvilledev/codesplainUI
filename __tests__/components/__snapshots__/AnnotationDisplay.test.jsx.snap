exports[`<AnnotationDisplay /> matches snapshot 1`] = `
<div>
  <MarkdownRenderer
    markdown="some annotation"
    options={
      Object {
        "highlight": [Function],
        "langPrefix": "hljs language-",
      }
    } />
  <RaisedButton
    disabled={false}
    fullWidth={false}
    label="Close"
    labelPosition="after"
    onTouchTap={[Function]}
    primary={false}
    secondary={true} />
  <RaisedButton
    disabled={false}
    fullWidth={false}
    label="Edit"
    labelPosition="after"
    onTouchTap={[Function]}
    primary={true}
    secondary={false} />
</div>
`;
